### ISSUE TITLE
Form Block: Persistent Blur

### ISSUE THREAD

**Issue by carinapilar** [Automattic/jetpack#28883]
### Impacted plugin

Jetpack

### Quick summary

Form block on user's site is showing with a blur that's not being removed after loading.

### Steps to reproduce

I wasn't able to replicate the issue on a test site, tried on Simple and Atomic.

### Impact

One

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Simple, Atomic

### Logs or notes

Workaround was to add a CSS code to remove the blur from the Form blocks.

It seems this change was implemented here: https://github.com/Automattic/jetpack/pull/28473 - I'm reporting this in the hope of having a specific situation that could be investigated further by the team.

The issue is happening on the site from the user on this chat: 40509071-hc - it doesn't seem to be caused by a plugin, since the issue persists after deactivating all of them - also tried to re-save Global Styles, but no luck.

**Comment by ha-un** [comment]
Another case: 39957118-hc

Used CSS workaround:

```
.wp-block-jetpack-contact-form-container {
    filter: none !important;
}
```

**Comment by sdixon194** [comment]
cc @jcheringer, I believe you worked on the contact form features recently. Would you mind having a look?

**Comment by ckmccoy2** [comment]
another case: 39981798-hc

**Comment by jcheringer** [comment]
Thanks for pinging me @sdixon194!
I'll take a look.

**Comment by edequalsawesome** [comment]
Another case in #5945418-zen, provided the CSS workaround for now

**Comment by goblinartificer** [comment]
also in 38799417-hc using the Skatepark theme. I tested Skatepark's Form blocks on an AT and a simple test site, and could only replicate this on the AT site. on the simple site, `.contact-form-styles-loaded` actually kicks in on the body element, unblurring the form; on the AT site, it doesn't seem to be there _while AMP is active_.

**Comment by Robertght** [comment]
Another case in 5945523-zen

**Comment by edequalsawesome** [comment]
Reported in 5946257-zen -- provided the CSS for now.

**Comment by kelasante** [comment]
Another case in 40505926-hc. Shared CSS.

**Comment by jcheringer** [comment]
### Tl;DR

The issue seems to be caused by the concatenation of some scripts, allowed by the `Page Optimize` plugin.
To fix it, go to the plugin settings page:

![Screen Shot 2023-02-10 at 15 34 36](https://user-images.githubusercontent.com/7811225/218170621-0f30a293-4e9b-4bd7-b4bc-162561c84b56.png)

And add `contact-form-styles` to the list of excluded files:

![image](https://user-images.githubusercontent.com/7811225/218170802-31703a25-c552-41c9-952a-5661ab1e979f.png)

---

### Context

During our investigation, we noticed that the blur effect wasn't being removed because a required Forms script wasn't running on the page.
That happened because the `Page Optimize` plugin was concatenating the JS files to make the page load faster, but one of the concatenated scripts (All in One SEO plugin) was triggering a JS error, causing the entire file to fail to load.
You can solve the issue by disabling the script concatenation or adding `contact-form-styles` to the list of exceptions.

An optimized solution would be making the `contact-form-styles` default for the Page Optimize plugin, but that'd take more time.

**Comment by kraftbj** [comment]
Since `Page Optimize` is an a8c plugin that is provided by the WoA platform, we should shim it without requiring using action, IMO.

`$do_concat = apply_filters( 'js_do_concat', $do_concat, $handle );` ([source](https://github.com/Automattic/page-optimize/blob/master/concat-js.php#L167))

**Comment by jcheringer** [comment]
> Since `Page Optimize` is an a8c plugin that is provided by the WoA platform, we should shim it without requiring using action, IMO.
> 
> `$do_concat = apply_filters( 'js_do_concat', $do_concat, $handle );` ([source](https://github.com/Automattic/page-optimize/blob/master/concat-js.php#L167))

Oh... That's nice! Thanks for sharing @kraftbj 
That's for sure a better solution.
I'll prepare a PR for that.

**Comment by jcheringer** [comment]
@kraftbj, to confirm if I understood the filter correctly.
I've tried the following:

```PHP
add_filter( 'js_do_concat', '__return_false' );
```

But this seems to disable the concatenation for all JS scripts.
I was wondering if there's a way we could avoid just the `contact-form-styles` from being concatenated.

**Comment by kraftbj** [comment]
@jcheringer Since the `apply_filters` passes the `$handle`, we can do something akin to this (untested)

```
add_filter(
    'js_do_concat',
    function( $do_concat, $handle ) {
        if ( 'contact-form-styles' === $handle ) {
            $do_concat = false;
        }
        return $do_concat;
    },
    10,
    2
);
```

**Comment by jcheringer** [comment]
Thanks for the example @kraftbj 
The WP filters are still a bit mind-blowing for me 😅 
That worked flawlessly!

**Comment by Joshgrygiel** [comment]
Another: 38493116-hc

**Comment by nazilya06** [comment]
Another case: 5948626-hc

**Comment by hrn1995** [comment]
Another case: 5945717 - zen

However, they are on a Free plan, no workaround will work for them at the moment

**Comment by hrn1995** [comment]
Another case: 5948654 - zen

**Comment by gamebits** [comment]
This issue is affecting all sites hosted by Newspack, where AMP is required. See p1676050091374619-slack-C01C378QWRX

**Comment by happychait** [comment]
Another report in 5947338-zen

**Comment by nazilya06** [comment]
Another case: 5951766 - HC

**Comment by aisajib** [comment]
Some reports on the forum: https://wordpress.com/forums/topic/contact-form-is-all-blurred-out/

**Comment by csonnek** [comment]
Another report on the forums: https://wordpress.com/forums/topic/contact-form-not-displaying/?view=all

Edited to add this report is on the Free WordPress.com plan, so no access to CSS.

**Comment by csonnek** [comment]
Another forums report: https://wordpress.com/forums/topic/long-time-to-load-page/?view=all

Site on the Free WordPress.com plan

**Comment by sdixon194** [comment]
Thanks folks, a fix for this has been merged and will be released with 11.8.1 within the next 24-48 hours hopefully.

**Comment by goblinartificer** [comment]
and again in 21824729-hc, provided CSS as they didn't want to deactivate AMP and adding that file to Page Optimize didn't work. plus one more in 40599776-hc (simple site, does not happen for me, but user is on a very old MBP with an older version of Safari). 

@sdixon194 out of curiosity, does #28905 resolve this for cases where AMP is the issue and/or for simple sites? it seems specific to Page Optimize, and I'm wondering why we're blurring the form in the first place since there's other ways that can cause trouble.

**Comment by sophiegyo** [comment]
>@sdixon194 out of curiosity, does https://github.com/Automattic/jetpack/pull/28905 resolve this for cases where AMP is the issue and/or for simple sites? it seems specific to Page Optimize, and I'm wondering why we're blurring the form in the first place since there's other ways that can cause trouble.

@goblinartificer Judging from the reports we've seen in the forum the fix does work for Simple sites; I checked one now and the form's not appearing blurry at this time. I have no idea why though!

**Comment by ghost** [comment]
Another case: 5950184-zen provided the CSS workaround.

**Comment by jcheringer** [comment]
@goblinartificer, #28905 just solves the issue with the Page Optimize plugin.
#28926 should fix the cases where AMP is the issue.

> and I'm wondering why we're blurring the form in the first place since there's other ways that can cause trouble.

Recently, we've implemented some changes to improve the Forms styles, and that change depends on a JS script.
To prevent the user from seeing their forms in a pre-styled version, we introduced the blur effect as a loading state for the Form.
Due to some AMP limitations, we're completely removing the blur effect from these sites.
For the other sites, the blur effect should appear just for a brief moment.

**Comment by upwardmomentum84** [comment]
Another case: 34151412-hc provided the CSS workaround.

**Comment by christianguerrag** [comment]
I'm re aading these tickets that are actually from zen:
5948626-zen
5945717-zen
5948654-zen
5951766-zen

**Comment by supernovia** [comment]
No CSS workaround for this person:
https://wordpress.com/forums/topic/form-pages-arent-working-on-my-site/

Do we have an ETA on a fix or any other workaround for no-CSS sites? This is kinda big.

**Comment by supernovia** [comment]
Noting too on this site:
https://mainewestrentals.wordpress.com/signing-a-lease/

It looks fine for staff, but not for logged-out users

@jcheringer can you have a look at that link in incognito?

**Comment by filipanoscampos** [comment]
Another case here: 40398435-hc
Shared CSS workaround.

Follow up here 5960499-zd-woothemes

**Comment by Robertght** [comment]
5958419-zen

**Comment by jcheringer** [comment]
Hi @supernovia! A fix is on the way.

https://github.com/Automattic/jetpack/pull/28973 should fix the issue for slow-loading pages like the one you mentioned above.

**Comment by supernovia** [comment]
Ah thanks @jcheringer . I wonder what's making that page take so long to load... 🤔

**Comment by adelineyaw** [comment]
Another case here: 36967127-hc and 5949209-zen - shared CSS workaround

**Comment by cuemarie** [comment]
Another instance here: 34282898-hc

**Comment by ha-un** [comment]
Another case: 39295606-hc - shared CSS workaround.

**Comment by dereksmart** [comment]
Hi everyone, these 3 fixes were deployed to wpcom: 
This one should fix the AMP issue: #28926
This one for Page Optimize: #28905
This one for the slow loading: #28973

This should resolve for wpcom simple sites. 
Atomic will get fixed when this is deployed p9o2xV-2OJ-p2

**Comment by rw-ye** [comment]
Another report - 38989962-hc

**Comment by pauljacobson** [comment]
I have an instance of this issue in 5960347-zen 

In this case, the Atomic site uses the third party [Kadence](https://www.kadencewp.com/) theme and Kadence Blocks. I shared the Custom CSS workaround.

**Comment by masperber** [comment]
Another instance: 33083764-hc

Applied workaround with CSS.

**Comment by susanjanec** [comment]
Another instance - 38369146-hc

Provided CSS workaround.

**Comment by dereksmart** [comment]
Fixes were deployed to all environments! Should be resolved.

**Comment by helenaartmann** [comment]
Got another report of this still happening here: [6039972-zn](6039972-zd-woothemes)

**Comment by sdixon194** [comment]
Hmm as far as I understand, the blur was removed entirely in https://github.com/Automattic/jetpack/pull/29178 

I checked their sites and don't see that they've added any custom CSS as well. And the form doesn't load blurred for me on multiple browsers. If I'd have to guess, I'd assume things were cached somewhere.
